scripts/docsgen-cli/main.go

```diff
@@ -45,89 +46,21 @@ func main() {
 	}
 
 	fmt.Println("Generating CLI documentation...")
-	var eg errgroup.Group
-	for _, cmd := range []string{"lotus", "lotus-miner", "lotus-worker"} {
-		eg.Go(func() error {
-			err := generateMarkdownForCLI(cmd)
-			if err != nil {
-				fmt.Printf(" ❌ %s: %v\n", cmd, err)
-			} else {
-				fmt.Printf(" ✅ %s\n", cmd)
-			}
-			return err
-		})
-	}
-	if err := eg.Wait(); err != nil {
-		fmt.Printf("Failed to generate CLI documentation: %v\n", err)
-		os.Exit(1)
-	}
-	fmt.Println("Documentation generation complete.")
-}
-
-func generateMarkdownForCLI(cli string) error {
-	md := filepath.Join(outputDir, fmt.Sprintf("cli-%s.md", cli))
-	out, err := os.Create(md)
-	if err != nil {
-		return err
-	}
-	defer func() { _ = out.Close() }()
-	return writeCommandDocs(out, cli, 0)
-}
-
-func writeCommandDocs(file *os.File, command string, depth int) error {
-	// For sanity, fail fast if depth exceeds some arbitrarily large number. In which
-	// case, chances are there is a bug in this script.
-	if depth > depthRecursionLimit {
-		return fmt.Errorf("recursion exceeded limit of %d", depthRecursionLimit)
-	}
 
-	// Get usage from the command.
-	usage, err := exec.Command("sh", "-c", "./"+command+" -h").Output()
-	if err != nil {
-		return fmt.Errorf("failed to run '%s': %v", command, err)
+	cliApps := map[string]*cli.App{
+		"lotus":        lotus.App(),
+		"lotus-worker": worker.App(),
+		"lotus-miner":  miner.App(),
 	}
 
-	// Skip the first new line since the docs do not start with a newline at the very
-	// top.
-	if depth != 0 {
-		if _, err := file.WriteString("\n"); err != nil {
-			return err
+	for name, app := range cliApps {
+		generator := NewDocGenerator(outputDir, app)
```

Comment:
```suggestion
	for name, app := range cliApps {
		// reset HelpName manually because shared Command objects get modified
		// by the first app they are added to
		for _, cmd := range app.Commands {
			cmd.HelpName = fmt.Sprintf("%s %s", app.HelpName, cmd.Name)
		}
		generator := NewDocGenerator(outputDir, app)
```

this is all that's needed to reset the problem that you're working around by making the `Command`s in functions: https://github.com/urfave/cli/blob/v2-maint/app.go#L232-L240

Unfortunately you can't just reset `HelpName` and re-run `app.Setup()` because it's a strictly run-once function, which is pretty annoying. But since we don't do anything sophisticated in `HelpName` I think it's good enough to just rewrite it here. Then you can undo all of the functions you had to make (sorry), but hopefully it's just a case of checking thise files out from master again.

Reply:
actually probably wont' be that easy because you had to put them in a new package, but hopefully it's straightforward enough to just un-`func` those shared cmds